### PR TITLE
#2559. Add augmenting functions signature tests. Part 1

### PR DESCRIPTION
**Pull request by sgrekhov**

### CODE REVIEW

File: LanguageFeatures/Augmentation-libraries/augmenting_functions_A04_t01_lib.dart [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+// [cfe] unspecified
+  augment static String? staticMethod3() => "42";
+//                       ^^^^^^^^^^^^^
+// [analyzer] unspecified
+// [cfe] unspecified
+  augment static String staticMethod4<T extends String>(T t) => "42";
+//                      ^^^^^^^^^^^^^
+// [analyzer] unspecified
+// [cfe] unspecified
+
```
Comment (eernstg) [inline LanguageFeatures/Augmentation-libraries/augmenting_functions_A04_t01_lib.dart:199]: Missing `staticMethod4`?

File: LanguageFeatures/Augmentation-libraries/augmenting_functions_A04_t01_lib.dart [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+// [cfe] unspecified
+  augment String? instanceMethod3() => "42";
+//                ^^^^^^^^^^^^^^^
+// [analyzer] unspecified
+// [cfe] unspecified
+  augment String instanceMethod4<T extends String>(T t) => "42";
+//               ^^^^^^^^^^^^^^^
+// [analyzer] unspecified
+// [cfe] unspecified
+}
```
Comment (eernstg) [inline LanguageFeatures/Augmentation-libraries/augmenting_functions_A04_t01_lib.dart:216]: Missing `instanceMethod5`.

File: LanguageFeatures/Augmentation-libraries/augmenting_functions_A04_t08_lib.dart [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+// [cfe] unspecified
+  augment static void staticMethod2([int o]) {}
+//                    ^^^^^^^^^^^^^
+// [analyzer] unspecified
+// [cfe] unspecified
+  augment static void staticMethod3([String? s]) {}
+//                    ^^^^^^^^^^^^^
+// [analyzer] unspecified
+// [cfe] unspecified
+  augment static void staticMethod3([String? s]) {}
```
Comment (eernstg) [inline LanguageFeatures/Augmentation-libraries/augmenting_functions_A04_t08_lib.dart:57]: Duplicate by accident?

File: LanguageFeatures/Augmentation-libraries/augmenting_functions_A04_t08_lib.dart [hunk 4]
```diff
@@ -0,0 +1,236 @@
...
```
Comment (eernstg) [inline LanguageFeatures/Augmentation-libraries/augmenting_functions_A04_t08_lib.dart:105]: Duplicate by accident?

File: LanguageFeatures/Augmentation-libraries/augmenting_functions_A04_t08_lib.dart [hunk 5]
```diff
@@ -0,0 +1,236 @@
...
```
Comment (eernstg) [inline LanguageFeatures/Augmentation-libraries/augmenting_functions_A04_t08_lib.dart:155]: Duplicate by accident?

File: LanguageFeatures/Augmentation-libraries/augmenting_functions_A04_t08_lib.dart [hunk 6]
```diff
@@ -0,0 +1,236 @@
...
```
Comment (eernstg) [inline LanguageFeatures/Augmentation-libraries/augmenting_functions_A04_t08_lib.dart:203]: Dup by accident?

**Review by eernstg — COMMENTED** [review]
Looks good. I commented on a few declarations that seem to be duplicates of previous declarations for no clear reason.

File: LanguageFeatures/Augmentation-libraries/augmenting_functions_A04_t01_lib.dart (hunk 1 above)
Comment (sgrekhov) [reply LanguageFeatures/Augmentation-libraries/augmenting_functions_A04_t01_lib.dart:203]: Yes, `staticMethod5` was missing. Added

File: LanguageFeatures/Augmentation-libraries/augmenting_functions_A04_t01_lib.dart (hunk 2 above)
Comment (sgrekhov) [reply LanguageFeatures/Augmentation-libraries/augmenting_functions_A04_t01_lib.dart:224]: Added

File: LanguageFeatures/Augmentation-libraries/augmenting_functions_A04_t08_lib.dart (hunk 3 above)
Comment (sgrekhov) [reply LanguageFeatures/Augmentation-libraries/augmenting_functions_A04_t08_lib.dart]: Really. Removed

File: LanguageFeatures/Augmentation-libraries/augmenting_functions_A04_t08_lib.dart (hunk 4 above)
Comment (sgrekhov) [reply LanguageFeatures/Augmentation-libraries/augmenting_functions_A04_t08_lib.dart]: Removed

File: LanguageFeatures/Augmentation-libraries/augmenting_functions_A04_t08_lib.dart (hunk 5 above)
Comment (sgrekhov) [reply LanguageFeatures/Augmentation-libraries/augmenting_functions_A04_t08_lib.dart]: Removed

File: LanguageFeatures/Augmentation-libraries/augmenting_functions_A04_t08_lib.dart (hunk 6 above)
Comment (sgrekhov) [reply LanguageFeatures/Augmentation-libraries/augmenting_functions_A04_t08_lib.dart]: Removed

**Review by sgrekhov — COMMENTED** [review]
Thank you! Duplicated declarations removed

**Review by eernstg — APPROVED** [review]

**Pull request merged**